ast/edit/match.go

```diff
@@ -0,0 +1,436 @@
+package edit
+
+import (
+	"fmt"
+
+	"github.com/influxdata/flux/ast"
+)
+
+// Match takes an AST and a pattern and returns the nodes of the AST that
+// match the given pattern.
+// The pattern is an AST in turn, but it is partially specified: nil nodes are ignored
+// while matching, and slices are checked for the first matched sub-sequence.
```

Comment:
Can you give an example of how slice matching works?

Reply:
I wanted to change slice matching behavior, actually:

```
Two slices match if every node provided in the pattern exists in the slice subject to comparison.
```

I think it is what you really want when thinking of "matching" and the code for the comparison would also be more readable. What do you think?

Reply:
Maybe.  An empty slice in a pattern would match anything, which could be the opposite of what you want.  I suppose you could use the following:
- pattern is `nil` slice, matches anything
- pattern is empty slice, matches only an empty slice
- pattern is non-empty slice, matches slices that are a superset of pattern

This is still inconsistent though.  I'm not sure the best way to handle this issue,